### PR TITLE
Fix NoCloud docs:

### PR DESCRIPTION
**Pull request by holmanb**
```
- fix failing example command
- augment "alternative commands" with more common utilities
- drop reference to decade old release
```

qemu-img correctly identifies the format of the backing file, but fails anyways:
```
arc/tmp qemu-img create -f qcow2 -b disk.img boot-disk.img 
qemu-img: boot-disk.img: Backing file specified without backing format
Detected format of qcow2.%                                                                                            arc/tmp echo $?
1
```

Fix by adding `-F qcow2` to invocation.

Additionally, add some context to the `mcopy` command, as well as alternative option with mount/umount/cp

### CODE REVIEW

File: doc/rtd/topics/datasources/nocloud.rst [hunk 1]
```diff
@@ -61,35 +61,42 @@ You may also optionally provide a vendor-data file in the following format.
...
 
-    ## create a new qcow image to boot, backed by your original image
-    $ qemu-img create -f qcow2 -b disk.img boot-disk.img
+    ## 2b) option 1: mount and copy files
+    ## $ sudo mount -t vfat /tmp/seed.img /mnt
+    ## $ sudo cp user-data meta-data /mnt
+    ## $ sudo umount /mnt
+
+    ## 2b) option 2: the mtools package provides mcopy, which can access vfat
+    filesystems without mounting them
```
Comment (TheRealFalcon) [inline doc/rtd/topics/datasources/nocloud.rst:87]: Missing the `##` here.

File: doc/rtd/topics/datasources/nocloud.rst [hunk 2]
```diff
@@ -61,35 +61,42 @@ You may also optionally provide a vendor-data file in the following format.
...
+    ## $ mcopy -oi seed.img user-data meta-data
+
+    ## 3) create a new qcow image to boot, backed by your original image
+    $ qemu-img create -f qcow2 -b disk.img -F qcow2 -K qcow2 boot-disk.img
```
Comment (TheRealFalcon) [inline doc/rtd/topics/datasources/nocloud.rst:91]: `qemu-img: unrecognized option '-K'`

Even fixing that, I can't get this command to run.
```
qemu-img create -f qcow2 -b disk.img -F qcow2 boot-disk.img 
qemu-img: boot-disk.img: Could not open 'disk.img': No such file or directory
Could not open backing image.
```

I followed 2b option 1.

Comment (TheRealFalcon) [inline doc/rtd/topics/datasources/nocloud.rst:91]: `qemu-img: unrecognized option '-K'`

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Instructions don't work for me. See my inline comments.

Additionally, since we're here, can we also update:
```bash
## 1) create user-data and meta-data files that will be used
## to modify image on first boot
$ { echo instance-id: iid-local01; echo local-hostname: cloudimg; } > meta-data
$ printf "#cloud-config\npassword: passw0rd\nchpasswd: { expire: False }\nssh_pwauth: True\n" > user-data
```

Seems a bit silly to write to files using echo for one and printf for the other.

File: doc/rtd/topics/datasources/nocloud.rst (hunk 2 above)
Comment (holmanb) [reply doc/rtd/topics/datasources/nocloud.rst:91]: The instructions start with:
```
Given a disk ubuntu cloud image in 'disk.img', you can create a sufficient disk by following the example below.
```

Comment (TheRealFalcon) [reply doc/rtd/topics/datasources/nocloud.rst:91]: Yes, I did. I suppose I should rename my image. :smile:

File: doc/rtd/topics/datasources/nocloud.rst [hunk 3]
```diff
@@ -61,35 +61,42 @@ You may also optionally provide a vendor-data file in the following format.
...
 
-    ## boot the image and login as 'ubuntu' with password 'passw0rd'
+    ## 4) boot the image and login as 'ubuntu' with password 'passw0rd'
     ## note, passw0rd was set as password through the user-data above,
     ## there is no password set on these images.
     $ kvm -m 256 \
        -net nic -net user,hostfwd=tcp::2222-:22 \
        -drive file=boot-disk.img,if=virtio \
-       -drive file=seed.iso,if=virtio
+       -drive driver=raw,file=seed.iso,if=virtio
```
Comment (TheRealFalcon) [inline doc/rtd/topics/datasources/nocloud.rst:99]: We named our seed `seed.img` earlier but are referencing `seed.iso` here.

**Comment by holmanb** [comment]
@TheRealFalcon Thanks for the review! I think I addressed all of your comments.

**Review by TheRealFalcon — APPROVED** [review]
Looks good, thanks!

**Pull request merged**